src/ucp/rma/rma_send.c

```diff
@@ -133,8 +133,9 @@ ucp_rma_request_init(ucp_request_t *req, ucp_ep_h ep, const void *buffer,
     req->send.ep              = ep;
     req->send.buffer          = (void*)buffer;
     req->send.datatype        = ucp_dt_make_contig(1);
-    req->send.mem_type        = ucp_request_get_memory_type(context, buffer,
-                                                            length, param);
+    req->send.mem_type        = ucp_request_get_memory_type(
+                                    context, buffer, length, req->send.datatype,
```

Comment:
ucp_dt_make_contig(1) instead of req->send.datatype
or we could use local helper var for it..
let compiler optimize